src/configure/clients/github/githubClient.ts

```diff
@@ -36,41 +36,38 @@ export class GithubClient {
         await this._setGithubSecret(secretName, secretKeyObject.key_id, encryptedEncodedText);
     }
 
-    public async createGithubRepo(orgName: string, repoName: string): Promise<void | GitHubRepo> {
+    public async createGithubRepo(orgName: string, repoName: string): Promise<GitHubRepo> {
         let restClient = new RestClient();
         let Url = "https://api.github.com/orgs/" + orgName + "/repos";
-        try {
-            return restClient.sendRequest(<UrlBasedRequestPrepareOptions>{
-                url: Url,
-                headers: {
-                    "Content-Type": "application/json",
-                    "Authorization": "Bearer " + this.patToken,
-                    "User-Agent": UserAgent
-                },
-                method: 'POST',
-                body: {
-                    "name": repoName,
-                    "description": "Repo created from VScode extension 'Deploy to Azure'",
-                    "homepage": "https://github.com",
-                    "private": false,
-                    "has_issues": true,
-                    "has_projects": true,
-                    "has_wiki": true
-                },
-                deserializationMapper: null,
-                serializationMapper: null
-            })
-                .then((detail: GitHubRepo) => {
-                    return detail;
-                }).catch(error => {
-                    telemetryHelper.logError(Layer, TracePoints.GitHubRepositoryCreationFailed, error);
+        return restClient.sendRequest3(<UrlBasedRequestPrepareOptions>{
+            url: Url,
+            headers: {
+                "Content-Type": "application/json",
+                "Authorization": "Bearer " + this.patToken,
+                "User-Agent": UserAgent
+            },
+            method: 'POST',
+            body: {
+                "name": repoName,
+                "description": "Repo created from VScode extension 'Deploy to Azure'",
+                "homepage": "https://github.com",
+                "private": true,
+                "has_issues": true,
+                "has_projects": true,
+                "has_wiki": true
+            },
+            deserializationMapper: null,
+            serializationMapper: null
+        })
+            .then((detail: GitHubRepo) => {
+                return detail;
+            }).catch(error => {
+                telemetryHelper.logError(Layer, TracePoints.GitHubRepositoryCreationFailed, error);
+                if (error.response.statusCode === 422) {
```

Comment:
when does this happen?

Reply:
It happens when repo creation fails due to the existence of another repo of same name.